plugin/hooks.go

```diff
@@ -155,4 +156,6 @@ type Hooks interface {
 	// Note that this method will be called for files uploaded by plugins, including the plugin that uploaded the post.
 	// FileInfo.Size will be automatically set properly if you modify the file.
 	FileWillBeUploaded(c *Context, info *model.FileInfo, file io.Reader, output io.Writer) (*model.FileInfo, string)
+
+	ReceivePluginEvent(c *Context, event string, payload interface{})
```

Comment:
I think something along the lines of `OnPluginEvent` may be more fitting. Looking at the pattern of naming, it seems appropriate to me. A comment needs to be placed here as well. Something like

> OnPluginEvent is invoked on each subscribed plugin when a PluginEvent is dispatched.

Also, I think we should define a `PluginEvent` struct to represent the event, with the actual data payload as an `interface{}` inside of the event.